src/agents/gc/OSConfig_PrototypeResource.c

```diff
@@ -0,0 +1,658 @@
+    MI_UNREFERENCED_PARAMETER(resourceClass);
+
+    OSConfig_PrototypeResource_TestTargetResource test_result_object;
+
+    MI_Result miResult = MI_RESULT_OK;
+    MI_Boolean is_compliant = MI_FALSE;
+
+    MI_Value miValueResult;
+    memset(&miValueResult, 0, sizeof(MI_Value));
+
+    OsConfigLogInfo(GetLog(), "TestTargetResource: start (namespace %s, class name %s, method name %s)", nameSpace, className, methodName);
+
+    if ((in == NULL) || (in->InputResource.exists == MI_FALSE) || (in->InputResource.value == NULL))
+    {
+        OsConfigLogError(GetLog(), "TestTargetResource: invalid OSConfig_PrototypeResource_TestTargetResource argument");
+        miResult = MI_RESULT_FAILED;
+        goto Exit;
+    }
+
+    if ((in->InputResource.value->PrototypeClassKey.exists == MI_FALSE) &&  (in->InputResource.value->PrototypeClassKey.value != NULL))
+    {
+        OsConfigLogError(GetLog(), "TestTargetResource: no PrototypeClassKey");
+        miResult = MI_RESULT_FAILED;
+        goto Exit;
+    }
+
+    if (strncmp(g_prototypeClassKey, in->InputResource.value->PrototypeClassKey.value, ARRAY_SIZE(g_prototypeClassKey) - 1))
+    {
+        // Refresh our stored key
+        memset(g_prototypeClassKey, 0, sizeof(g_prototypeClassKey));
+        strncpy(g_prototypeClassKey, in->InputResource.value->PrototypeClassKey.value, ARRAY_SIZE(g_prototypeClassKey) - 1);
+    }
+
+    if ((in->InputResource.value->Ensure.exists == MI_TRUE) && (in->InputResource.value->Ensure.value != NULL))
+    {
+        if ((!strcmp(in->InputResource.value->Ensure.value, "Present")) || (!strcmp(in->InputResource.value->Ensure.value, "Absent")))
+        {
+            memset(g_ensure, 0, sizeof(g_ensure));
+            memcpy(g_ensure, in->InputResource.value->Ensure.value, ARRAY_SIZE(g_ensure) - 1);
+
+            // Simulating compliance either way
+            is_compliant = MI_TRUE;
+        }
+        else
+        {
+            OsConfigLogError(GetLog(), "TestTargetResource: unknown ensure value (%s)", in->InputResource.value->Ensure.value);
+            is_compliant = MI_FALSE;
+        }
+    }
+    
+    if (MI_RESULT_OK != (miResult = OSConfig_PrototypeResource_TestTargetResource_Construct(&test_result_object, context)))
+    {
+        OsConfigLogError(GetLog(), "TestTargetResource: OSConfig_PrototypeResource_TestTargetResource_Construct failed with %d", miResult);
+        goto Exit;
+    }
+
+    if (MI_RESULT_OK != (miResult = OSConfig_PrototypeResource_TestTargetResource_Set_MIReturn(&test_result_object, 0)))
+    {
+        OsConfigLogError(GetLog(), "TestTargetResource: OSConfig_PrototypeResource_TestTargetResource_Set_MIReturn failed with %d", miResult);
+        goto Exit;
+    }
+
+    OSConfig_PrototypeResource_TestTargetResource_Set_Result(&test_result_object, is_compliant);
+    MI_Context_PostInstance(context, &(test_result_object.__instance));
+
+    miResult = MI_RESULT_OK;
+    MI_Context_PostError(context, miResult, MI_RESULT_TYPE_MI, 0);

```

Comment:
Did you intend to always call `MI_Context_PostError`? I realize this is is work in progress. Just checking.

Reply:
Good catch